### PR TITLE
Bug 1882572 - Instrumenting Translations Nimbus Feature Release

### PR DESCRIPTION
**Pull request by ohall-m**
This patch adds instrumentation for releasing separate sections of the translations feature.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1882572

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt [hunk 1]
```diff
@@ -302,12 +302,10 @@ class BrowserFragment : BaseBrowserFragment(), UserInteractionHandler {
...
             context.components.core.store.state.translationEngine.isEngineSupported
-        if (
-            !context.settings().enableTranslations &&
-            (
-                isEngineSupported == null ||
-                    isEngineSupported == false
-                )
+
+        if (isEngineSupported != true ||
+            !context.settings().enableTranslations ||
```
Comment (ohall-m) [inline fenix/app/src/main/java/org/mozilla/fenix/browser/BrowserFragment.kt:307]: Removing the secret settings `!context.settings().enableTranslations` in bug [1853058](https://bugzilla.mozilla.org/show_bug.cgi?id=1853058).

File: fenix/app/nimbus.fml.yaml [hunk 2]
```diff
@@ -375,6 +375,50 @@ features:
...
         default: 1
 
+  translations:
+    description: The feature that allows on-device translations of web content.
+    variables:
+      main-flow-toolbar-enabled:
+        description: >
+          Show the primary toolbar entry point into the translations feature. (Translations icon on URL toolbar.)
+        type: Boolean
+        default: true
```
Comment (ohall-m) [inline fenix/app/nimbus.fml.yaml:385]: Will flip the Nimbus default to false in bug [1853058](https://bugzilla.mozilla.org/show_bug.cgi?id=1853058). (Having it as a default of `true` while secret settings is still around, but the `main-flows` should probably be `false` before release.)

File: fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationOptionsDialog.kt [hunk 3]
```diff
@@ -65,14 +66,16 @@ fun TranslationOptionsDialog(
...
 
-        item {
-            TextListItem(
-                label = stringResource(id = R.string.translation_option_bottom_sheet_translation_settings),
-                modifier = Modifier
-                    .fillMaxWidth()
-                    .padding(start = 56.dp),
-                onClick = { onTranslationSettingsClicked() },
-            )
+        if (FxNimbus.features.translations.value().globalSettingsEnabled) {
```
Comment (gabrielluong) [inline fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationOptionsDialog.kt:69]: I am expecting this to break all the Compose Previews. My recommendation here is to add a suitably named parameter for this composable (eg, `showTranslationSettings = true`) where we pass in these Nimbus features.

Comment (ohall-m) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationOptionsDialog.kt:69]: It has a compile-time default and the Compose Previews render for me. WDYT?

image

Comment (gabrielluong) [reply fenix/app/src/main/java/org/mozilla/fenix/translations/TranslationOptionsDialog.kt:69]: ok no problem from me then.

**Review by gabrielluong — APPROVED** [review]

**Pull request merged**